extra/cloudformation/default-unauth-resources-template.yaml

```diff
@@ -208,9 +219,77 @@ Resources:
   AmazonLocationDemoKinesisBucket:
     Condition: EnablePinTranslate
     Type: 'AWS::S3::Bucket'
-    DeletionPolicy: Retain
+    DeletionPolicy: Delete
     Properties:
       BucketName: !Sub "location.aws.com.demo.kinesis.${AWS::AccountId}"
+      VersioningConfiguration:
+          Status: Enabled
+      BucketEncryption:
+          ServerSideEncryptionConfiguration:
+           - ServerSideEncryptionByDefault:
+              SSEAlgorithm: aws:kms
+
+  AmazonLocationDemoKinesisBucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !Ref AmazonLocationDemoKinesisBucket
+      PolicyDocument:
+        Version: "2012-10-17"
+        Id: PutObjPolicy
+        Statement:
+          - Sid: DenyIncorrectEncryptionHeader
+            Effect: Deny
+            Principal: "*"
+            Action: "s3:PutObject"
+            Resource: !Sub "arn:aws:s3:::${AmazonLocationDemoKinesisBucket}/*"
+            Condition:
+              'StringNotEquals':
+                s3:x-amz-server-side-encryption: 'aws:kms'
+          - Sid: DenyUnEncryptedObjectUploads
+            Effect: Deny
+            Principal: "*"
+            Action: "s3:PutObject"
+            Resource: !Sub "arn:aws:s3:::${AmazonLocationDemoKinesisBucket}/*"
+            Condition:
+              'Null':
+                s3:x-amz-server-side-encryption: 'true'
+
+  AmazonLocationDemoKMS:
+    Type: AWS::KMS::Key
+    Properties:
+      Description: You created this resource by running Amazon Location Demo's CloudFormation template.
+      KeyPolicy:
+        Version: '2012-10-17'
+        Id: default-policy
+        Statement:
+          - Sid: Enable IAM User Permissions
+            Effect: Allow
+            Principal:
+              AWS: !Sub arn:aws:iam::${AWS::AccountId}:root
+            Action: kms:*
+            Resource: '*'
+          - Sid: Allow KMS Key Administrators to Manage Key
+            Effect: Allow
+            Principal:
+              AWS: 
+              - !GetAtt CognitoIdentityPoolUnauthRole.Arn
+              - !GetAtt AmazonLocationKinesisDeliveryRole.Arn
+            Action:
+              - kms:Create*
+              - kms:Describe*
+              - kms:Enable*
+              - kms:List*
+              - kms:Put*
+              - kms:Update*
+              - kms:Revoke*
+              - kms:Disable*
+              - kms:Get*
+              - kms:Delete*
+              - kms:TagResource
+              - kms:UntagResource
+              - kms:ScheduleKeyDeletion
+              - kms:CancelKeyDeletion
```

Comment:
Can you elaborate a bit more on why unauth user need KMS permission? And if it does, does it need all above permissions?

Reply:
@harshit-makeen can you kindly answer @vinguthh, thanks.

Reply:
@vinguthh resolved